stages/org.osbuild.coreos.platform

```diff
@@ -0,0 +1,115 @@
+   are primarily console settings.
+4. Apply any platform specific kernel arguments along with the
+   `ignition.platform.id={platform-name}` kernel argument.
+5. Create the grub console.cfg file and apply any platform
+   specific grub console configuration.
+This stage is highly CoreOS specific and subject to change in the
+future if/when we change the way platform specific information is
+defined in our broader efforts to share more defaults with OSBuild.
+"""
+SCHEMA_2 = r"""
+"options": {
+  "additionalProperties": false,
+  "properties": {
+    "platform": {
+      "description": "The target platform name/ID",
+      "type": "string"
+    }
+  }
+},
+"devices": {
+  "type": "object",
+  "additionalProperties": true
+},
+"mounts": {
+  "type": "array"
+}
+"""
+import json
+import os
+import osbuild.api
+import shutil
+import sys
+
+from osbuild.util import bls
+
+# Constants
+#For console.cfg see https://github.com/coreos/bootupd/pull/620
+#For platforms.json see https://github.com/coreos/coreos-assembler/pull/3709
+GRUB_CONSOLE_CFG = "boot/grub2/console.cfg"
+PLATFORMS_JSON_DEST= "boot/coreos/platforms.json"
+PLATFORMS_JSON_SOURCE = "usr/share/coreos-assembler/platforms.json"
+
+def generate_console_settings_file(console_settings, file_path):
+    settings_content = ""
+    if console_settings is not None:
+        settings_content = "\n".join(console_settings)
+    file_content = f"""
+# Any non-default console settings will be inserted here.
+# CONSOLE-SETTINGS-START
+{settings_content}
+# CONSOLE-SETTINGS-END
+"""
+    with open(file_path, 'w') as file:
+        file.write(file_content)
+
+
+def process_platforms_json(json_file_path, platform):
+    keys = ["grub_commands", "kernel_arguments"]
+    result = {}
+    with open(json_file_path, 'r') as file:
+        data = json.load(file)
+        if platform in data:
+            for key in keys:
+                if key in data[platform]:
+                    result[key] = data[platform][key]
+
+    return result.get("grub_commands", []),\
+           result.get("kernel_arguments", [])
+
+
+def main(paths, tree, options):
+
+    deployment = options.get("deployment")
+    platform = options.get("platform")
+
+    root = args["paths"]["mounts"]
+    boot_path = os.path.join(root, "boot")
+    platforms_source_path = os.path.join(root, PLATFORMS_JSON_SOURCE.lstrip())
+    platforms_dest_path = os.path.join(root, PLATFORMS_JSON_DEST.lstrip())
+    grub_console_cfg_path = os.path.join(root, GRUB_CONSOLE_CFG.lstrip())
+
+    kernel_arguments = [f"ignition.platform.id={platform}"]
+
+    json_grub_args, json_kargs = None, None
+    if os.path.exists(platforms_source_path):
+        os.makedirs(os.path.dirname(platforms_dest_path), mode=0o755, exist_ok=True)
+        # Copy platforms.json to the boot partition
+        shutil.copy2(platforms_source_path, platforms_dest_path)
+        json_grub_args, json_kargs = process_platforms_json(platforms_dest_path, platform)
```

Comment:
(super niptick) At this point the json is processed, right? so this is a list of extra_grub_args and extra_kargs, correct? maybe drop the json here and just use something like extra_ as a prefix or just no prefix (in the sense that the fact that they came from json input is not super important anymore at this point :)

Reply:
either way works. I kind of liked it being more explicit because we also have `kernel_arguments` var and knowing which one came from the json is nice.

Reply:
Thanks! Yeah, sorry, that was an incrediable nitpick, totally fine to keep of course